docs/topics/type-safe-builders.md

```diff
@@ -242,7 +242,46 @@ html {
 }
 ```
 
-## Full definition of the com.example.html package
+You can also apply the `@DslMarker` annotation directly to [function types](lambdas.md#function-types).
+To do so, annotate a `@DslMarker` annotation with `@Target(AnnotationTarget.TYPE)`:
+
+```kotlin
+@Target(AnnotationTarget.TYPE)
+@DslMarker
+annotation class HtmlTagMarker
+```
+
+This allows the `@DslMarker` annotation to be applied to function types, most commonly to lambdas with receivers. For example:
+
+```kotlin
+fun HTML.head(init: @HtmlTagMarker Head.() -> Unit): Head {
```

Comment:
NB: I do not know what is the preferred way of doing code examples in our technical documentation, you are the expert here and your opinion is the most important =)

I would prefer to have a more complete example w.r.t. `@DslMarker`, meaning we should also define the `html` function

```
fun html(init: @HtmlTagMarker HTML.() -> Unit): HTML {
    val html = HTML()
    html.init()
    return html
}
```

Reply:
Also, maybe we could keep the bodies of the functions empty or as `TODO()`? Their definitions are not that important w.r.t. DSL scoping rules, and the ones we have now are kinda not entirely correct.